akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst

```diff
@@ -168,4 +168,19 @@ As users were often confused by "when to use an Actor vs. when to use an Agent?"
 the Agents, as they rarely are really enough and do not fit the Akka spirit of thinking about distribution.
 We also anticipate to replace the uses of Agents by the upcoming Akka Typed, so in preparation thereof the Agents have been deprecated in 2.5.
 
-If you use Agents and would like to take over the maintanance thereof, please contact the team on gitter or github.
\ No newline at end of file
+If you use Agents and would like to take over the maintanance thereof, please contact the team on gitter or github.
+
+Distributed Data
+================
+
+Map allow generic type for the keys
+-----------------------------------
+
+In 2.4 the key of any Distributed Data map always needed to be of type String. In 2.5 you can use any type for the key. This means that
+every map (ORMap, LWWMap, PNCounterMap, ORMultiMap) now takes an extra type parameter to specify the key type. To migrate
+existing code from 2.4 to 2.5 you simple add String as key type, for example: `ORMultiMap[Foo]` becomes `ORMultiMap[String, Foo]`.
+`PNCounterMap` didn't take a type parameter in version 2.4, so `PNCounterMap` in 2.4 becomes `PNCounterMap[String]` in 2.5.
+Java developers should use `<>` instead of `[]`, e.g: `PNCounterMap<String>`.
+
+**NOTE: Even though the interface is not compatible between 2.4 and 2.5, the binary protocol over the wire is. This means that
+2.4 nodes can synchronize with 2.5 nodes.**
```

Comment:
It's only compatible as long as String keys are used, right? I guess it's obvious but might make sense to add it to this sentence.